classes/Hook/HookActionValidateOrder.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\Ps_Googleanalytics\Hooks;
+
+use Context;
+use Ps_Googleanalytics;
+
+class HookActionValidateOrder implements HookInterface
+{
+    /**
+     * @var Ps_Googleanalytics
+     */
+    private $module;
+
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @var array
+     */
+    private $params;
+
+    public function __construct(Ps_Googleanalytics $module, Context $context)
+    {
+        $this->module = $module;
+        $this->context = $context;
+    }
+
+    /**
+     * run
+     *
+     * @return void
+     */
+    public function run()
+    {
+        // Check if we are creating backoffice order
+        if ($this->context->controller->controller_name != 'AdminOrders' && $this->context->controller->controller_name != 'Admin') {
+            return;
+        }
+
+        // Mark this ID to immediately display it on next page load
+        $order = $this->params['order'];
+        if (!empty($this->context->cookie->__get('ga_admin_order'))) {
+            $ga_admin_order = $this->context->cookie->__get('ga_admin_order') . ',' . $order->id;
```

Comment:
Suggestion: use sprintf instead.

I personnally find it easier to read ;) and I think it's less prone to errors.

Reply:
Will do, using dots is me still being a beginner. :-)

Reply:
@Hlavtox you can also stop using `__get` or `__set`, you can just do 

`$this->context->cookie->ga_admin_order` which calls `__get`

or `$this->context->cookie->ga_admin_order = 'something'` which calls `__set` 👍🏻

Reply:
@kpodemski Fixed